### ISSUE TITLE
non-headless mode: page closed when interacting with devtools console

### ISSUE THREAD

**Issue by gnuletik** [go-rod/rod#770]
With the following program:

```go
package main

import (
	"os"
	"os/signal"
	"syscall"

	"github.com/go-rod/rod"
	"github.com/go-rod/rod/lib/launcher"
)

func main() {
	u := launcher.New().Headless(false).MustLaunch()
	rod.New().ControlURL(u).MustConnect().MustPage("https://wikipedia.org")

	c := make(chan os.Signal, 1)
	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
	<-c
}
```

`Rod Version: v0.112.2`

When:

- Starting `go run .`
- Opening devtools console (F12)
- Typing a character (e.g. `c`)

The page gets closed with no error message and the go program is still running.

I tried enabling Trace, and changing a few launch parameters without any success.

Do you know why this is happening?

Thanks!

**Comment by gnuletik** [comment]
The issue does not occurs when running the go program mentioned above on GNU/Linux.
However, it occurs on MacOS (tested on 13.0.1).

I added:

```
l.Logger(os.Stdout)
```

I can see that the crash comes from Chromium browser

```
[1203/155730.424635:WARNING:crash_report_exception_handler.cc(235)] UniversalExceptionRaise: (os/kern) failure (5)
```

I can reproduce the issue by running Chromium with rod's parameters:

```
~/.cache/rod/browser/chromium-1075642/chrome-mac/Chromium.app/Contents/MacOS/Chromium [...]
```

But the issue does not occurs when using my host's Chromium instance, when using the same arguments:

```
/Applications/Chromium.app/Contents/MacOS/Chromium [...]
```

rod's chromium version is `Version 110.0.5439.0`.
My host's chromium version is `Version 106.0.5229.0`

I checked puppeteer releases and found out that they are currently using Chromium 108.0.5351.0 (r1056772), see https://github.com/puppeteer/puppeteer/releases/tag/v19.2.0.

If I set the same version with `go-rod`:

```
l.Revision(1056772)
```

The issue is gone :)

I checked the Chromium release dashboard here: https://chromiumdash.appspot.com/schedule
And the current stable release is Chromium v108.

I don't think that rod should use Chromium v110, which is unstable.
https://github.com/go-rod/rod/blob/master/lib/launcher/revision/main.go

@ysmood are you open to change the default Chromium revision to the latest stable instead of the latest of every release? It could be useful to follow Puppeteer versions which is maintained by the Chrome DevTools team.

Thanks!

**Comment by ysmood** [comment]
The current revision is based on playwright,  If you check the source code of https://github.com/go-rod/rod/blob/master/lib/launcher/revision/main.go

Playwright is also maintained by the chromium team,  It's almost the same developers of puppeteer.

Yes, I'm open to change anything.

I'm using the latest Mac OS m1, but I don't have any problem. Can you give more details about your platform?

**Comment by ysmood** [comment]
Also read the code here, it's already using the same way puppeteer download chromium, rod and puppeteer use the same npm mirror address:

https://github.com/go-rod/rod/blob/17b8fba7c0a32493491528b15d9749d83a128bb3/lib/launcher/revision/main.go#L19

**Comment by gnuletik** [comment]
Thanks for the fast feedback !

When checking the latest release of Playwright here: https://github.com/microsoft/playwright/releases/tag/v1.28.1
It says that Chromium `108.0.5359.29` is used.
But `go-rod` is using Chromium `109.0.5392.0`.

It seems that the playwright version is only used on Linux/ARM.
https://github.com/go-rod/rod/blob/20d37c378c595a4244d0870925320637e8a491be/lib/launcher/browser.go#L60-L69

Here are the specs of my laptop:
MacBook Pro 2021 M1 Pro, 32GB RAM, MacOS Ventura 13.0.1

Here is a full reproduction of the issue.
![2022-12-04 16 06 36](https://user-images.githubusercontent.com/9059840/205498733-a64ff55c-c511-4c66-bec3-0efd3260c707.gif)

**Comment by ysmood** [comment]
If you check their actual source code of v1.28.1:

https://raw.githubusercontent.com/microsoft/playwright/v1.28.1/packages/playwright-core/browsers.json

The version of chromium is 1033, which is 109.0.5392.0, I think it's their bug. You can download it yourself with their cdn.

**Comment by ysmood** [comment]
Thanks for you research, I can reproduce it now.

Maybe should just use the cdn of playwright as the default revision for chromium, could you make a PR to change the automation file?

**Comment by gnuletik** [comment]
Great that the issue is reproducible on your side too!

I just tried to download the following release (r1033) from Playwright builds: https://playwright.azureedge.net/builds/chromium/1033/chromium-mac-arm64.zip

Chrome About sections says `Version 108.0.5359.29`

However, the crash is also present on this version (even without running it through `rod`).

The weird part is that the `Google Chrome` installed on my host says `Version 108.0.5359.94 (Official Build) (arm64)` but there's no issue.
So, maybe a fix was implemented on latest 108.x versions ?
I'm not familiar with Chromium / Google Chrome releases cycle.

So, I don't think that using playwright as the source of truth would help :/
But I'm open to make a PR once we found out a good source of truth!
